### PR TITLE
fix: no grace period

### PR DESCRIPTION
**Pull request by Godley**

### CODE REVIEW

File: kube-scheduler-healthcheck [hunk 1]
```diff
@@ -18,7 +18,7 @@ endOfPodDef
 
 alias kubectl="timeout -t 30 kubectl --namespace \$TEST_POD_NAMESPACE"
  
-kubectl delete pod $TEST_POD_NAME || true
+kubectl delete pod $TEST_POD_NAME --grace-period=0 || true
```
Comment (mikebryant) [inline kube-scheduler-healthcheck:21]: Pondering here. If the node it's on is dead (broken docker, node shut down etc), this still won't do it.

Maybe we should be doing something like:
```
delete pod --grace-period=0
sleep 1
delete pod --grace-period=0 --force
```

Comment (Godley) [reply kube-scheduler-healthcheck:21]: should we `|| true` all of those?

Comment (mikebryant) [reply kube-scheduler-healthcheck:21]: Ah, yes

File: kube-scheduler-healthcheck [hunk 2]
```diff
@@ -18,7 +18,11 @@ endOfPodDef
...
+sleep 1
+kubectl delete pod --grace-period=0 --force || true
 kubectl create -f /tmp/pod.yaml
 until [ "$(kubectl get ${TEST_POD_NAME} -o jsonpath={.status.phase})" != "Pending" ]; do sleep 1; done
-kubectl delete pod $TEST_POD_NAME
+kubectl delete pod $TEST_POD_NAME --grace-period=0
+sleep 1
+kubectl delete pod --grace-period=0 --force
```
Comment (Godley) [inline kube-scheduler-healthcheck]: @mikebryant wondering whether to `|| true` here as well?

Comment (mikebryant) [reply kube-scheduler-healthcheck]: Yes, we'll need to

**Review by mikebryant — APPROVED** [review]

**Pull request merged**